server/src/main/java/org/opensearch/rest/ProtectedRoute.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest;
+
+/**
+ * A named Route
+ *
+ * @opensearch.api
+ */
+public class ProtectedRoute extends RestHandler.Route {
```

Comment:
@cwperks sorry for coming late, do you think `NamedRoute` would be a better name? Protected (to me)  sounds like something we have to handle cautiously but in fact this is just a way to bundle actions and permissions together (and not necessarily protect), just an option.

Reply:
Thank you for the feedback @reta ! I will switch this over to `NamedRoute` and add validation. I've addressed this comment on my local branch, but I'm not able to push to github atm and it looks like there is a status advisory: https://www.githubstatus.com/